### PR TITLE
Support creating tables with table comment in Delta Lake

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Support creating tables with table comment in Delta Lake. Verified the Delta Lake's compatibility with Spark locally. 

## Documentation

(x) No documentation is needed.

## Release notes

(x) Release notes entries required with the following suggested text:

```markdown
# Delta Lake
* Support storing a table comment when creating new tables. ({issue}`12452`)
```

### CODE REVIEW

**Comment by findepi** [comment]
> Support creating tables with table comment in Delta Lake & Kudu. Verified the Delta Lake's compatibility with Spark locally.

Remove Kudu from the description ?

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeDatabricksConnectorTest.java [hunk 1]
```diff
@@ -13,11 +13,36 @@
...
 {
     public TestDeltaLakeDatabricksConnectorTest()
     {
         super("databricks-test-queries", "io/trino/plugin/deltalake/testing/resources/databricks/");
     }
+
+    @Override
+    public void testShowCreateTable()
+    {
+        // Table comment is different from TestDeltaLakeOssDeltaLakeConnectorTest
```
[the remark this answers is not in the document]
Comment (ebyhr) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeDatabricksConnectorTest.java]: As we talked offline, I suppose `CheckpointEntryIterator#getString()` needs fix to handle `null` value correctly. Just pushed additional commit.

**Comment by ebyhr** [comment]
Rebased on upstream to resolve conflicts.

**Comment by ebyhr** [comment]
Added a product test. Ready for review now.

**Review by findinpath — APPROVED** [review]

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeOssCreateTable.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import static io.trino.tests.product.TestGroups.DELTA_LAKE_DATABRICKS;
+import static io.trino.tests.product.TestGroups.DELTA_LAKE_OSS;
+import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.trino.tests.product.hive.util.TemporaryHiveTable.randomTableSuffix;
+import static io.trino.tests.product.utils.QueryExecutors.onDelta;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.lang.String.format;
+import static org.testng.Assert.assertEquals;
+
+public class TestDeltaLakeOssCreateTable
```
[the remark this answers is not in the document]
Comment (ebyhr) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeOssCreateTable.java]: Moved to `TestDeltaLakeDatabricksCreateTableCompatibility` and tested locally.

**Comment by ebyhr** [comment]
CI hit #12300

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/transactionlog/checkpoint/TestCheckpointEntryIterator.java [hunk 3]
```diff
@@ -87,8 +87,8 @@ public void testReadMetadataEntry()
                 .isEqualTo(
                         new MetadataEntry(
                                 "b6aeffad-da73-4dde-b68e-937e468b1fde",
-                                "",
-                                "",
+                                null,
```
Comment (findepi) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/transactionlog/checkpoint/TestCheckpointEntryIterator.java:90]: is "name" a table name? shouldn't it be present?

**Review by findepi — APPROVED** [review]

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java [hunk 4]
```diff
@@ -521,8 +521,12 @@ private DeltaLakeTransactionLogEntry buildTxnEntry(ConnectorSession session, Blo
         return DeltaLakeTransactionLogEntry.transactionEntry(result);
     }
 
+    @Nullable
     private String getString(Block block, int position)
     {
+        if (block.isNull(position)) {
+            return null;
+        }
```
[the remark this answers is not in the document]
Comment (ebyhr) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java:529]: This is not new support. It returned an empty character before this change. 
`TestCheckpointEntryIterator` covers both non-empty and null cases. 

Added `checkArgument` to those three methods.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/transactionlog/checkpoint/TestCheckpointEntryIterator.java (hunk 3 above)
Comment (ebyhr) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/transactionlog/checkpoint/TestCheckpointEntryIterator.java:90]: It's optional "User-provided identifier for this table" according to https://github.com/delta-io/delta/blob/master/PROTOCOL.md#change-metadata. 

I can send another PR if you agree on setting the field at https://github.com/trinodb/trino/blob/master/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java#L1004

**Pull request merged**